### PR TITLE
[ExportVerilog] Add lowering option to not emit version comment

### PR DESCRIPTION
**Pull request by rsetaluri**

### CODE REVIEW

**Comment by rsetaluri** [comment]
A few notes:

* It definitely sounds better to have the positive option, `emitVersionComment` instead of the negative `doNotEmitVerisonComment`, but it either requires allowing passing in a `=true`/`=false` option rather than just the option as a positive, since we *do* want to emit the version comment by default.
* In light of ^, not sure if `doNotEmitVersionComment` is the right naming convention

**Review by uenoku — APPROVED** [review]
LGTM. Could you update https://github.com/llvm/circt/blob/main/docs/VerilogGeneration.md? RE: naming, perhaps `omitVersionComment` might be better but I think `doNotEmitVersionComment`  also makes sense.

**Comment by rsetaluri** [comment]
> LGTM. Could you update https://github.com/llvm/circt/blob/main/docs/VerilogGeneration.md? RE: naming, perhaps `omitVersionComment` might be better but I think `doNotEmitVersionComment` also makes sense.

done! thanks :)

File: lib/Support/LoweringOptions.cpp
Comment (dtzSiFive) [inline lib/Support/LoweringOptions.cpp:117]: I think this should have an entry in `toString` as well (like the other options)?

**Review by dtzSiFive — APPROVED** [review]
LGTM.

It'd be nice to have a test for this -- we indirectly test the normal behavior (most directly in `test/Conversion/ExportVerilog/output-file.mlir`) but a small test for on/off seems nice.
LMK if you'd like to add this and need any help or have questions, otherwise something for us to add later :+1: .

See comment about also updating the `toString` method.

File: lib/Support/LoweringOptions.cpp
Comment (rsetaluri) [reply lib/Support/LoweringOptions.cpp:117]: done

File: include/circt/Support/LoweringOptions.h
Comment (darthscsi) [inline include/circt/Support/LoweringOptions.h:156]: We try to avoid negatives here (but haven't been perfectly consistent, c.f. disallowExpressionInliningInPorts).  Please use emitVersionComment.

**Review by darthscsi — APPROVED** [review]
Should change the sense in the LoweringOptions structure, but otherwise LGTM.

File: include/circt/Support/LoweringOptions.h
Comment (rsetaluri) [reply include/circt/Support/LoweringOptions.h:156]: See this comment: https://github.com/llvm/circt/pull/4811#issuecomment-1465321751

I think by default we want this on to avoid everyone having who is expecting this to explicitly enable this. I agree the negative is not great, but it seems like a better option than changing the default opaquely?

Comment (mwachs5) [reply include/circt/Support/LoweringOptions.h:156]: `omitVersionComment` would satisfy both right?

Comment (rsetaluri) [reply include/circt/Support/LoweringOptions.h]: done! now `omitVersionComment`. thanks for the feedback all :)

**Comment by rsetaluri** [comment]
@darthscsi @dtzSiFive please let me know if this can be merged now. thanks!

**Comment by rsetaluri** [comment]
> @darthscsi @dtzSiFive please let me know if this can be merged now. thanks!

Hi, hoping this PR can be merged soon. Thanks!

**Comment by teqdruid** [comment]
You've got the approvals, so you're good to go. Approval + comments means you're good to go and we trust you to make the changes requested.

Also, avoid amended, force-pushed commits after people start reviewing. Makes it difficult to see what you've changed in the case where a re-review is necessary.

**Comment by rsetaluri** [comment]
> You've got the approvals, so you're good to go. Approval + comments means you're good to go and we trust you to make the changes requested.
> 
> Also, avoid amended, force-pushed commits after people start reviewing. Makes it difficult to see what you've changed in the case where a re-review is necessary.

Thanks. Will avoid force-pushing in the future.

It looks like I can't actually perform the merge though? I don't have write access.

**Pull request merged**
